testing/ef-tests/src/cases/blockchain_test.rs

```diff
@@ -90,50 +91,47 @@ impl Case for BlockchainTestCase {
-                {
-                    let static_file_provider = provider.static_file_provider();
-                    let mut receipts_writer =
-                        static_file_provider.latest_writer(StaticFileSegment::Receipts).unwrap();
-                    receipts_writer.increment_block(0).unwrap();
-                    receipts_writer.commit_without_sync_all().unwrap();
-                }
-
                 // Decode and insert blocks, creating a chain of blocks for the test case.
-                let last_block = case.blocks.iter().try_fold(None, |_, block| {
+                let mut blocks = Vec::with_capacity(case.blocks.len());
+                for block in &case.blocks {
                     let decoded =
                         SealedBlock::<reth_primitives::Block>::decode(&mut block.rlp.as_ref())?;
-                    provider.insert_historical_block(decoded.clone().try_recover().unwrap())?;
-                    Ok::<Option<SealedBlock>, Error>(Some(decoded))
-                })?;
-                provider
-                    .static_file_provider()
-                    .latest_writer(StaticFileSegment::Headers)
-                    .unwrap()
-                    .commit_without_sync_all()
-                    .unwrap();
+                    let recovered_block = decoded.clone().try_recover().unwrap();
 
-                // Execute the execution stage using the EVM processor factory for the test case
-                // network.
-                let _ = ExecutionStage::new_with_executor(
-                    reth_evm_ethereum::execute::EthExecutorProvider::ethereum(chain_spec.clone()),
-                    Arc::new(EthBeaconConsensus::new(chain_spec)),
-                )
-                .execute(
-                    &provider,
-                    ExecInput { target: last_block.as_ref().map(|b| b.number), checkpoint: None },
-                );
+                    provider.insert_block(recovered_block.clone(), StorageLocation::Database)?;
+
+                    blocks.push(recovered_block);
+                }
+                let last_block = blocks.last().cloned();
+
+                // Initialize executor with state
+                let executor_provider = EthExecutorProvider::ethereum(chain_spec);
+                for block in blocks {
+                    let state_provider = provider.history_by_block_hash(block.parent_hash)?;
+                    let state_db = StateProviderDatabase(state_provider);
+                    let executor = executor_provider.executor(state_db);
+
+                    // Execute all blocks in a batch
+                    if let Ok(output) = executor.execute(&block) {
+                        provider.write_state(
+                            &ExecutionOutcome::single(block.number, output),
+                            OriginalValuesKnown::Yes,
+                            StorageLocation::Database,
+                        )?;
+                    }
```

Comment:
@mattsse / @rkrasiuk Any insights as to why the following doesn't work?

Changing `let state_provider = LatestStateProviderRef::new(&provider)` makes the tests pass, however I am trying to execute each block using the same code pattern as debug_executionWitness; ie get the state of the chain when the parent block was executed and execute the child block using that state.